nixos/modules/services/networking/sozu.nix

```diff
@@ -0,0 +1,299 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sozu;
+  format = pkgs.formats.toml {};
+
+  settings = format.generate "settings.toml" cfg.settings;
+
+  configFile = pkgs.writeText "config.toml" ''
+    ${fileContents settings}
+    ${optionalString (cfg.extraConfig != null) cfg.extraConfig}
+  '';
+in
+{
+  options.services.sozu = {
+    enable = mkEnableOption "Sozu server";
+
+    settings = mkOption {
+      default = {};
+      type = types.submodule {
+
+        options = {
+          saved_state = mkOption {
+            default = "/run/sozu/state.json";
+            type = types.str;
```

Comment:
```suggestion
            type = types.file;
```

Reply:
`types.file` doesn't exist, and `types.path`is not what I want here.
`types.str` has been choses intentionally.

Reply:
Is this still a concern I am missing the point of, or can I resolve this?

Reply:
I'll be addressing this point later anyways, so you can ignore for now.